syntax = "proto3";

option csharp_namespace = "NowWeTalkin.Server.Grpc";
option java_package = "br.com.prodatamobility.grpc.nowWeTalkin";
option go_package = "prodatamobility.com.br/grpc/nowwetalkin";

import "google/protobuf/timestamp.proto";

package operations;

service Hotlist {
  rpc RequestHotlistUnary (HotlistRequest) returns (HotlistReply);
  rpc RequestHotlistServerStream (HotlistRequest) returns (stream HotlistReply);
}

message HotlistRequest {
  int64 index = 1;
}

message HotlistReply {
  int32 status = 1;
  int64 index = 2;
  int32 issuer_id = 3;
  int32 card_design_id = 4;
  uint32 card_serial_number = 5;
  HotlistAction action = 6;
  google.protobuf.Timestamp register_date = 7;
}

enum HotlistAction {
  UPDATE = 0;
  INSERT = 1;
  REMOVE = 2;
}